src/sql/workbench/contrib/notebook/test/browser/outputProcessor.test.ts

```diff
@@ -0,0 +1,164 @@
+import { getRandomElement } from 'vs/base/common/arrays';
+
+suite('OutputProcessor functions', function (): void {
+	const text = 'An arbitrary text input:!@#$%^&*()_+~`:;,.-_=';
+
+	const arbitraryMetadata = {
+		// arbitrarily construction chart options. It is any JSONObject
+		azdata_chartOptions: {
+			scale: true,
+			dataGrid: false,
+			arbitraryCharProperty: {
+				prop1: 'value1',
+				prop2: {
+					deepProp1: 3
+				}
+			}
+		}
+	};
+
+	suite('getData', function (): void {
+		// data tests
+		for (const outputType of ['execute_result', 'display_data', 'update_display_data'] as nbformat.OutputType[]) {
+			const output = <nbformat.IExecuteResult>{
+				output_type: outputType,
+				data: <op.IMimeBundle>{
+					result: text  // free form fields of the IMimeBundle. This is just an arbitrarily cooked-up hardcoded example for the test.
+				}
+			};
+			test(`test for outputType:${output.output_type}`, () => {
+				verifyGetDataForDataOutput(output);
+			});
+		}
+
+		// error tests
+		for (const traceback of [undefined, [undefined], [], [''], ['tb1:!@#$%^&*()_+~`:;,.-_=', 'tb2:!@#$%^&*()_+~`:;,.-_=']]) {
+			for (const evalue of [undefined, '', 'evalue1:!@#$%^&*()_+~`:;,.-_=']) {
+				for (const ename of ['ename1:!@#$%^&*()_+~`:;,.-_=', '', undefined]) {
+					const output = <nbformat.IError>{
+						output_type: 'error',
+						ename: ename,
+						evalue: evalue,
+						traceback: traceback
+					};
+					test(`test for outputType:'${output.output_type}', ename:'${ename}', evalue:${evalue}, and traceback:${JSON.stringify(traceback)}`, () => {
+						verifyGetDataForErrorOutput(output);
+					});
+				}
+			}
+		}
+
+		// stream tests
+		for (const name of ['stderr', 'stdout'] as nbformat.StreamType[]) {
+			const output = <nbformat.IStream>{
+				output_type: 'stream',
+				text: text,
+				name: name
+			};
+			test(`test for outputType:'${output.output_type} and name:${name}`, () => {
+				verifyGetDataForStreamOutput(output);
+			});
+		}
+	});
+
+	suite('getMetadata', function (): void {
+		for (const outputType of ['execute_result', 'display_data', getRandomElement(['update_display_data', 'stream', 'error'])] as nbformat.OutputType[]) {
+			const output: nb.ICellOutput = {
+				output_type: outputType
+			};
+			test(`test for outputType:${outputType}`, () => {
+				output.metadata = arbitraryMetadata;
+				const result = op.getMetadata(output);
+				if (nbformat.isExecuteResult(output) || nbformat.isDisplayData(output)) {
+					assert.deepEqual(result, output.metadata, `getMetadata should return the metadata object passed in the output object`);
+				} else {
+					assert.deepEqual(result, {}, `getMetadata should return an empty object when output_type is not IDisplayData or IExecuteResult`);
+				}
+			});
+		}
+	});
+
+	suite('getBundleOptions', function (): void {
+		for (const trusted of [true, false]) {
+			const outputType = getRandomElement(['execute_result', 'display_data']) as nbformat.OutputType;
```

Comment:
Hmm let's not introduce an element of chance here. If we'd like to test all scenarios, then let's not randomly pick an element and have this be deterministic.

Reply:
fixed